### PR TITLE
Simplified Qt sinks

### PR DESCRIPTION
**Pull request by mguludag**
Thanks @FlorianWolters for [this](https://github.com/gabime/spdlog/pull/1986#issuecomment-886739045) idea. New header is more simpler and no private classes derived from QObject. Its also thread-safe (logging from any thread is safe)

### CODE REVIEW

**Review by mguludag — COMMENTED** [review]
trim newline chars instead of remove 2 chars

File: include/spdlog/sinks/qt_sinks.h [hunk 1]
```diff
@@ -1,200 +1,129 @@
...
+
+//
+// qtextedit_sink class
+//
+namespace spdlog {
+namespace sinks {
+template <typename Mutex>
+class qtextedit_sink : public base_sink<Mutex> {
+public:
+  qtextedit_sink(QTextEdit *textedit = nullptr) {
```
Comment (gabime) [inline include/spdlog/sinks/qt_sinks.h:27]: why allow nullptr by default if it throws if nullptr?

File: include/spdlog/sinks/qt_sinks.h [hunk 2]
```diff
@@ -1,200 +1,129 @@
...
+        qtextedit = textedit;
+    } else {
+      throw spdlog_ex("Error opening QTextEdit");
+    }
+  }
+
+  ~qtextedit_sink() { flush_(); }
+
+protected:
+  void sink_it_(const details::log_msg &msg) override {
```
Comment (gabime) [inline include/spdlog/sinks/qt_sinks.h:38]: can never be null. why check it?

File: include/spdlog/sinks/qt_sinks.h [hunk 3]
```diff
@@ -1,200 +1,129 @@
...
+    memory_buf_t formatted;
+    base_sink<Mutex>::formatter_->format(msg, formatted);
+    string_view_t str = string_view_t(formatted.data(), formatted.size());
+    QMetaObject::invokeMethod(qtextedit,"append", Qt::AutoConnection, Q_ARG(QString, QString::fromUtf8(str.data(), static_cast<int>(str.size())).trimmed()));
+  }
+
+  void flush_() override {}
+
+private:
+    QTextEdit* qtextedit = nullptr;
```
Comment (gabime) [inline include/spdlog/sinks/qt_sinks.h:50]: please rename to qtextedit_ like rest of spdlog

**Review by gabime — CHANGES_REQUESTED** [review]

**Review by mguludag — COMMENTED** [review]
removed nullptr checks and renamed member vars

**Comment by gabime** [comment]
Thanks. Please fix the conflict so i can merge

File: include/spdlog/sinks/qt_sinks.h [hunk 4]
```diff
@@ -1,125 +1,125 @@
...
+
+  ~qtextedit_sink() { flush_(); }
+
+protected:
+  void sink_it_(const details::log_msg &msg) override {
+    memory_buf_t formatted;
+    base_sink<Mutex>::formatter_->format(msg, formatted);
+    string_view_t str = string_view_t(formatted.data(), formatted.size());
+    QMetaObject::invokeMethod(qtextedit_,"append", Qt::AutoConnection, Q_ARG(QString, QString::fromUtf8(str.data(), static_cast<int>(str.size())).trimmed()));
+  }
```
Comment (gabime) [inline include/spdlog/sinks/qt_sinks.h:43]: why the change from the simple append()? seems efficiant

Comment (mguludag) [reply include/spdlog/sinks/qt_sinks.h:43]: Direct calling GUI widget functions from another threads in Qt app results crash. Qt forces us to use their own Q_OBJECT macro and connect function or QMetaObject::invokemethod function.

**Comment by gabime** [comment]
Thanks @mguludag
